source/Events/UpdateWorldEntityPositionsEvent.cs

```diff
@@ -2,24 +2,25 @@
 using Game.Entities;
 using Game.Scenes.World;
 using System.Collections.Generic;
+using System.Linq;
 using EventArgs = Annex.Events.EventArgs;
 
 namespace Game.Events
 {
     public class UpdateWorldEntityPositionsEvent : CustomEvent
     {
         private readonly WorldScene _scene;
-        private readonly IList<BaseEntity> _entities;
+        private readonly IList<Entity> _entities;
 
-        public UpdateWorldEntityPositionsEvent(WorldScene scene, IList<BaseEntity> entities, string eventID, int interval_ms, int delay_ms) : base(eventID, interval_ms, delay_ms)
+        public UpdateWorldEntityPositionsEvent(WorldScene scene, IList<Entity> entities, string eventID, int interval_ms, int delay_ms) : base(eventID, interval_ms, delay_ms)
         {
             _scene = scene;
             _entities = entities;
         }
 
         protected override void Run(EventArgs args)
         {
-            foreach (var moveable in _entities)
+            foreach (var moveable in _entities.ToList())
```

Comment:
Is IList not already iterable? Why convert to a list?

Reply:
The foreach loop throws an exception because the list is being modified. This is because the player is being unloaded from the list when we change scene.